miner/src/transaction_queue.rs

```diff
@@ -335,12 +353,24 @@ impl TransactionQueue {
 				tx.hash(), tx.gas_price, self.minimal_gas_price
 			);
 
-			return Err(Error::Transaction(TransactionError::InsufficientGasPrice{
+			return Err(Error::Transaction(TransactionError::InsufficientGasPrice {
 				minimal: self.minimal_gas_price,
 				got: tx.gas_price,
 			}));
 		}
 
+		if tx.gas > self.gas_limit {
```

Comment:
But there already is similar check (or executive will just fail with `BlockGasLimitReached`).
In this PR we want to avoid even storing such transactions in TransactionQueue (`self.gas_limit here is 1.1*best_block_gas_limit).